sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -11,6 +11,7 @@ public abstract partial class BinaryContent : System.IDisposable
     {
         protected BinaryContent() { }
         public static System.ClientModel.BinaryContent Create(System.BinaryData value) { throw null; }
+        public static System.ClientModel.BinaryContent Create(System.IO.Stream stream) { throw null; }
```

Comment:
So we decided to go back to Create terminology (away from "From"). And the reason is to allow RequestContent inherit from BinaryContent? I like the benefit of being able to have the inheritance hierarchy, but there is a non-trivial drawback: I think it would be good to add an overload that takes a file path. Such overload would look as follows: Create(string filePath). But than, it creates dangerous inconsistency where BinaryContent.Create(someString) and BinaryData.Create(someString) have very different semantics and so a user can accidentally upload a file when they though they would be sending a string literal. 

Yet such filePath overload would be super useful because it allows efficient and reliable retries. The Stream overload allows retries only if the stream is seekable.

Reply:
That is fair.  Yes, as we discussed offline, we have a trade-off to make here between (1) allowing the Azure.Core inheritance hierarchy without RequestContent having duplicate methods or needing to EBN things to "remove" these vs (2) aligning better with BinaryData APIs and making the addition of a FromPath method consistent with other static factory methods on BinaryContent.

It sounded like we were sufficiently concerned with the additions to the Azure.Core type that this might be the better approach.  A rename is easy to implement though, at least until we GA the type.

If we did go with Create to allow the inheritance hierarchy, how would you feel about CreateFromPath for the file path overload you flagged?

If that's not great, how bad would it be to use EBN all over RequestContent to hide the duplicate methods?

Reply:
Also - it sounds like a request to add the file stream overload as well as part of this PR?  I wasn't aware that these weren't usually seekable, but let me go learn more about that ...

Reply:
It looks like FileStreams will often be seekable: 
> `true` if the stream supports seeking; `false` if the stream is closed or if the `FileStream` was constructed from an operating-system handle such as a pipe or output to the console.
https://learn.microsoft.com/en-us/dotnet/api/system.io.filestream.canseek?view=net-8.0#property-value

But let me look into what it would take to support a non-seekable stream in BinaryContent

Reply:
Discussed offline.  @KrzysztofCwalina shared that 1) it would be preferable for ClientModel APIs to align with BinaryData APIs, but 2) changing factory method names away from the Create pattern would mean we absolutely cannot inherit the Azure.Core RequestContent type from BinaryContent in the future.  My idea to [EBN] the BinaryContent APIs on RequestContent is not something we would consider because it breaks the inheritance paradigm -- i.e. it is bad to hide a method on the subtype that is present on the base type from an object-oriented perspective.  We might entertain the idea of later adding duplicate methods on BinaryContent to match the BinaryData ones and [EBN]'ing the Create methods if we were to never implement the RequestContent inheritance, but the benefit of that would likely not outweigh the cost of confusion to people using the Create APIs -- it wouldn't be a problem from an object-oriented perspective though.

Given this, since I think it is too high risk to put a stake in the ground and say "we will _never_ inherit RequestContent from BinaryContent" at this point in time -- given uncertainty about which approach we will take for Azure.Core 2.0 -- we will keep these names aligned with the Azure.Core type names for now, and if we want to add an overload that creates a BinaryContent from a file path, we will call it something like CreateFromPath.